### PR TITLE
Add support for closing the fab speed dial by tapping on the background

### PR DESCRIPTION
**Pull request by paour**
This requires setting up the FabSpeedDial as a full-screen component, and since the gravity is a bit strange, I had to add a spacer to make sure the FAB stays at the bottom.

### CODE REVIEW

**Comment by paour** [comment]
Oops, this also addresses #5

**Comment by paour** [comment]
Don't merge, this requires more time in the oven!

**Comment by paour** [comment]
OK, now ready for merging.

**Comment by yavski** [comment]
Thanks for looking into this @paour. What was wrong with the existing approach?

**Comment by paour** [comment]
The existing approach set a touchListener on the app's `content` ViewGroup, potentially stepping on the app's own listener. Additionally, touches were only detected if none of the contained views handled the touch (tapping on the background for example).

The approach of using a touchGuard also makes it possible to dim everything below the FAB when it opens (not implemented in this PR).

**Comment by yavski** [comment]
Ok, I'm happy to approve your PR - just update it please, you have merge conflicts.

File: library/src/main/java/io/github/yavski/fabspeeddial/FabSpeedDial.java [hunk 1]
```diff
@@ -258,25 +264,21 @@ public void onClick(View v) {
...
-                        removeFabMenuItems();
-                        return true;
-                    }
-                    return false;
-                }
-            });
+        if (useTouchGuard) {
+            ViewParent parent = getParent();
+
+            if (parent instanceof FrameLayout) {
```
Comment (yavski) [inline library/src/main/java/io/github/yavski/fabspeeddial/FabSpeedDial.java:285]: You can enhance the condition to check if `parent instanceof RelativeLayout`.

**Comment by paour** [comment]
Merged upstream

**Pull request merged**